tools/src/main/scala/scala/scalanative/codegen/llvm/compat/os/UnixCompat.scala

```diff
@@ -54,41 +49,21 @@ private[codegen] class UnixCompat(codegen: AbstractCodeGen)
     indent()
     line(s"$rec = $landingpad")
     line(s"$r0 = extractvalue $excRecTy $rec, 0")
-    line(s"$r1 = extractvalue $excRecTy $rec, 1")
-    line(s"$id = $typeid")
-    line(s"$cmp = icmp eq i32 $r1, $id")
-    line(s"br i1 $cmp, label %$excsucc, label %$excfail")
+    line(s"br label %$excsucc")
     unindent()
 
     line(s"$excsucc:")
     indent()
-    line(s"$w0 = call $ptrT $beginCatch($ptrT $r0)")
-    if (useOpaquePointers) {
-      line(s"$w2 = getelementptr ptr, ptr $w0, i32 1")
-      line(s"$exc = load ptr, ptr $w2")
-    } else {
-      line(s"$w1 = bitcast i8* $w0 to i8**")
-      line(s"$w2 = getelementptr i8*, i8** $w1, i32 1")
-      line(s"$exc = load i8*, i8** $w2")
-    }
-    line(s"call void $endCatch()")
-    str("br ")
+    line(s"$exc = call $ptrT $scalaNativeCatch($ptrT $r0)")
+    line("br ")
     codegen.genNext(next)
     unindent()
-
-    line(s"$excfail:")
-    indent()
-    line(s"resume $excRecTy $rec")
-    unindent()
```

Comment:
Since we catch all exceptions, this block can never happen.
Catching any exception that is not a Scala Native exception is undefined behavior